samples/dnn/dasiamrpn_tracker.py

```diff
@@ -0,0 +1,281 @@
+"""
+
+import numpy as np
+import cv2 as cv
+import argparse
+
+class DaSiamRPNTracker:
+    #initialization of used values, initial bounding box, used network
+    def __init__(self, im, target_pos, target_sz, net, kernel_r1, kernel_cls1):
+        self.windowing = "cosine"
+        self.exemplar_size = 127
+        self.instance_size = 271
+        self.total_stride = 8
+        self.score_size = (self.instance_size - self.exemplar_size) // self.total_stride + 1
+        self.context_amount = 0.5
+        self.ratios = [0.33, 0.5, 1, 2, 3]
+        self.scales = [8, ]
+        self.anchor_num = len(self.ratios) * len(self.scales)
+        self.penalty_k = 0.055
+        self.window_influence = 0.42
+        self.lr = 0.295
+        self.im_h = im.shape[0]
+        self.im_w = im.shape[1]
+        self.target_pos = target_pos
+        self.target_sz = target_sz
+        self.avg_chans = np.mean(im, axis=(0, 1))
+        self.net = net
+        self.score = []
+
+        if ((self.target_sz[0] * self.target_sz[1]) / float(self.im_h * self.im_w)) < 0.004:
+             raise AssertionError("Initializing BB is too small-try to restart tracker with larger BB")
+
+        self.anchor = self.__generate_anchor()
+        wc_z = self.target_sz[0] + self.context_amount * sum(self.target_sz)
+        hc_z = self.target_sz[1] + self.context_amount * sum(self.target_sz)
+        s_z = round(np.sqrt(wc_z * hc_z))
+
+        z_crop = self.__get_subwindow_tracking(im, self.exemplar_size, s_z)
+        z_crop = z_crop.transpose(2, 0, 1).reshape(1, 3, 127, 127).astype(np.float32)
+        self.net.setInput(z_crop)
+        z_f = self.net.forward('63')
+        kernel_r1.setInput(z_f)
+        r1 = kernel_r1.forward()
+        kernel_cls1.setInput(z_f)
+        cls1 = kernel_cls1.forward()
+        r1 = r1.reshape(20, 256, 4, 4)
+        cls1 = cls1.reshape(10, 256 , 4, 4)
+        self.net.setParam(self.net.getLayerId('65'), 0, r1)
+        self.net.setParam(self.net.getLayerId('68'), 0, cls1)
+
+        if self.windowing == "cosine":
+            self.window = np.outer(np.hanning(self.score_size), np.hanning(self.score_size))
+        elif self.windowing == "uniform":
+            self.window = np.ones((self.score_size, self.score_size))
+        self.window = np.tile(self.window.flatten(), self.anchor_num)
+
+    #creating anchor for tracking bounding box
+    def __generate_anchor(self):
+        self.anchor = np.zeros((self.anchor_num, 4),  dtype = np.float32)
+        size = self.total_stride * self.total_stride
+        count = 0
+
+        for ratio in self.ratios:
+            ws = int(np.sqrt(size / ratio))
+            hs = int(ws * ratio)
+            for scale in self.scales:
+                wws = ws * scale
+                hhs = hs * scale
+                self.anchor[count, 0] = 0
+                self.anchor[count, 1] = 0
+                self.anchor[count, 2] = wws
+                self.anchor[count, 3] = hhs
```

Comment:
I think is better to do: `self.anchor[count] = [0, 0, wws, hhs]`

Reply:
Done